internal/service/ec2/ec2_launch_template.go

```diff
@@ -1989,6 +1995,10 @@ func expandLaunchTemplateInstanceNetworkInterfaceSpecificationRequest(tfMap map[
 		apiObject.DeviceIndex = aws.Int64(int64(v))
 	}
 
+	if v, ok := tfMap["enable_primary_ipv6"].(bool); ok {
+		apiObject.PrimaryIpv6 = aws.Bool(v)
```

Comment:
Can somebody help me out a bit here?   I am getting an infinite drift a la:

```
      ~ network_interfaces {
          + enable_primary_ipv6 = "true"
            # (11 unchanged attributes hidden)
        }
```

I added a debug log line to this line and it logs that `enable_primary_ipv6` is indeed true

```
2024-03-17T09:11:44.743+0100 [DEBUG] aws_launch_template.web: applying the planned Update change
2024-03-17T09:11:44.747+0100 [DEBUG] provider.terraform-provider-aws: [DEBUG] ARIAN DEBUG tfMap: map[associate_carrier_ip_address: associate_public_ip_address: delete_on_termination: description: device_index:0 enable_primary_ipv6:true interface_type: ipv4_address_count:0 ipv4_addresses:0xc0018664c0 ipv4_prefix_count:0 ipv4_prefixes:0xc001866440 ipv6_address_count:1 ipv6_addresses:0xc001866500 ipv6_prefix_count:0 ipv6_prefixes:0xc001866480 network_card_index:0 network_interface_id: private_ip_address: security_groups:0xc001866540 subnet_id:]
```

But then the `CreateLaunchTemplateVersionRequest` seems to omit the `EnablePrimaryIpv6` field 

I have no idea why.

Is there some kind of weird  AWS SDK compatbility shim going on that might be dropping this newer field somewhere down the line? This is really strange. Or maybe a bug in `aws-go-sdk` itself?

```
  http.request.body=
  | Action=CreateLaunchTemplateVersion&ClientToken=terraform-20240317081144747000000001&LaunchTemplateData.BlockDeviceMapping.1.DeviceName=%2Fdev%2Fxvda&LaunchTemplateData.BlockDeviceMapping.1.Ebs.VolumeSize=4&LaunchTemplateData.IamInstanceProfile.Arn=arn%3Aaws%3Aiam%3A%3A686862074153%3Ainstance-profile%2Fweb&LaunchTemplateData.ImageId=ami-0da0e87f13dc26b38&LaunchTemplateData.InstanceType=t3.micro&LaunchTemplateData.KeyName=admin&LaunchTemplateData.NetworkInterface.1.DeviceIndex=0&LaunchTemplateData.NetworkInterface.1.Ipv6AddressCount=1&LaunchTemplateData.NetworkInterface.1.NetworkCardIndex=0&LaunchTemplateData.UserData=IyEvdXNyL2Jpbi9lbnYgYmFzaApzZXQgLWUKbml4IGJ1aWxkICcvbml4L3N0b3JlLzFqenYxY3JwYWszbjRqanIzaTNxZDdtZ2JyN2tqaGZ4LW5peG9zLXN5c3RlbS11bm5hbWVkLTIzLjExLjIwMjQwMzE0Ljg3OGVmN2QnIFwKICAtLXByb2ZpbGUgL25peC92YXIvbml4L3Byb2ZpbGVzL3N5c3RlbSBcCiAgLS1leHBlcmltZW50YWwtZmVhdHVyZXMgJ25peC1jb21tYW5kJyBcCiAgLS1leHRyYS1zdWJzdGl0dXRlcnMgJ3MzOi8vbml4b3MtdmlsbGFnZS1jYWNoZTIwMjQwMzE2MTAzMDUwMTM3NjAwMDAwMDAxP3JlZ2lvbj1ldS1jZW50cmFsLTEnIFwKICAtLWV4dHJhLXRydXN0ZWQtcHVibGljLWtleXMgJ25peG9zLXZpbGxhZ2U6NnlaMjZxb3BVWS9qRXJ0TS9ZdGpCMGpGc2hqeDRqZ1VZeEVtaHFLRWNRRT0nCi9uaXgvdmFyL25peC9wcm9maWxlcy9zeXN0ZW0vYmluL3N3aXRjaC10by1jb25maWd1cmF0aW9uIHN3aXRjaAo%3D&LaunchTemplateId=lt-08c9cfbf104b34b3a&Version=2016-11-15
```

Reply:
Yeh I'm pretty sure it's an issue with `aws-go-sdk`. when I print the go-sdk struct the field also disappears:

```
Yeh I'm pretty sure it's a bug in the `aws-go-sdk`:

When I log the input to `CreateLaunchTemplateVersionWithContext` the key also is not there

```
4-03-17T09:27:52.749+0100 [DEBUG] provider.terraform-provider-aws: [DEBUG] ARIAN: {
2024-03-17T09:27:52.749+0100 [DEBUG] provider.terraform-provider-aws:   ClientToken: "terraform-20240317082752748900000001",
2024-03-17T09:27:52.749+0100 [DEBUG] provider.terraform-provider-aws:   LaunchTemplateData: {
2024-03-17T09:27:52.749+0100 [DEBUG] provider.terraform-provider-aws:     BlockDeviceMappings: [{
2024-03-17T09:27:52.749+0100 [DEBUG] provider.terraform-provider-aws:         DeviceName: "/dev/xvda",
2024-03-17T09:27:52.749+0100 [DEBUG] provider.terraform-provider-aws:         Ebs: {
2024-03-17T09:27:52.749+0100 [DEBUG] provider.terraform-provider-aws:           VolumeSize: 4
2024-03-17T09:27:52.749+0100 [DEBUG] provider.terraform-provider-aws:         }
2024-03-17T09:27:52.749+0100 [DEBUG] provider.terraform-provider-aws:       }],
2024-03-17T09:27:52.749+0100 [DEBUG] provider.terraform-provider-aws:     IamInstanceProfile: {
2024-03-17T09:27:52.749+0100 [DEBUG] provider.terraform-provider-aws:       Arn: "arn:aws:iam::686862074153:instance-profile/web"
2024-03-17T09:27:52.749+0100 [DEBUG] provider.terraform-provider-aws:     },
2024-03-17T09:27:52.749+0100 [DEBUG] provider.terraform-provider-aws:     ImageId: "ami-0da0e87f13dc26b38",
2024-03-17T09:27:52.749+0100 [DEBUG] provider.terraform-provider-aws:     InstanceType: "t3.micro",
2024-03-17T09:27:52.749+0100 [DEBUG] provider.terraform-provider-aws:     KeyName: "admin",
2024-03-17T09:27:52.749+0100 [DEBUG] provider.terraform-provider-aws:     NetworkInterfaces: [{
2024-03-17T09:27:52.749+0100 [DEBUG] provider.terraform-provider-aws:         DeviceIndex: 0,
2024-03-17T09:27:52.749+0100 [DEBUG] provider.terraform-provider-aws:         Ipv6AddressCount: 1,
2024-03-17T09:27:52.749+0100 [DEBUG] provider.terraform-provider-aws:         NetworkCardIndex: 0
2024-03-17T09:27:52.749+0100 [DEBUG] provider.terraform-provider-aws:       }],
2024-03-17T09:27:52.749+0100 [DEBUG] provider.terraform-provider-aws:     UserData: <sensitive>
2024-03-17T09:27:52.749+0100 [DEBUG] provider.terraform-provider-aws:   },
2024-03-17T09:27:52.749+0100 [DEBUG] provider.terraform-provider-aws:   LaunchTemplateId: "lt-08c9cfbf104b34b3a"
2024-03-17T09:27:52.749+0100 [DEBUG] provider.terraform-provider-aws: }

```
```

Reply:
Fixed in https://github.com/hashicorp/terraform-provider-aws/pull/36425/commits/d2c1a993b1d792cd48d698cda11cb294fb98cf6b